bench/Main.hs

```diff
@@ -0,0 +1,283 @@
+      bench "documentSymbols" 100 $
+        fmap (either (not . null) (not . null)) . getDocumentSymbols,
+      ---------------------------------------------------------------------------------------
+      bench "documentSymbols after edit" 100 $ \doc -> do
+        let change =
+              TextDocumentContentChangeEvent
+                { _range = Just (Range (Position 854 23) (Position 854 23)),
+                  _rangeLength = Nothing,
+                  _text = " "
+                }
+        changeDoc doc [change]
+        either (not . null) (not . null) <$> getDocumentSymbols doc,
+      ---------------------------------------------------------------------------------------
+      bench "completions after edit" 10 $ \doc -> do
+        let change =
+              TextDocumentContentChangeEvent
+                { _range = Just (Range (Position 854 23) (Position 854 23)),
+                  _rangeLength = Nothing,
+                  _text = " "
+                }
+        changeDoc doc [change]
+        not . null <$> getCompletions doc (Position 853 12),
+      ---------------------------------------------------------------------------------------
+      benchWithSetup
+        "code actions"
+        10
+        ( \doc -> do
+            let p = Position 853 24
+            let change =
+                  TextDocumentContentChangeEvent
+                    { _range = Just (Range p p),
+                      _rangeLength = Nothing,
+                      _text = "a"
+                    }
+            changeDoc doc [change]
+            void (skipManyTill anyMessage message :: Session WorkDoneProgressEndNotification)
+            return p
+        )
+        ( \p doc -> do
+            not . null <$> getCodeActions doc (Range p p)
+        ),
+      ---------------------------------------------------------------------------------------
+      bench "code actions after edit" 10 $ \doc -> do
+        let p = Position 853 24
+        let change =
+              TextDocumentContentChangeEvent
+                { _range = Just (Range p p),
+                  _rangeLength = Nothing,
+                  _text = "a"
+                }
+        changeDoc doc [change]
+        void (skipManyTill anyMessage message :: Session WorkDoneProgressEndNotification)
+        not . null <$> getCodeActions doc (Range p p)
+    ]
+
+---------------------------------------------------------------------------------------------
+
+examplePackageName :: String
+examplePackageName = "Cabal"
+
+examplePackageVersion :: Version
+examplePackageVersion = makeVersion [3, 2, 0, 0]
+
+examplePackage :: String
+examplePackage = examplePackageName <> "-" <> showVersion examplePackageVersion
+
+exampleModulePath :: FilePath
+exampleModulePath = "Distribution" </> "Simple.hs"
+
+data Config = Config
+  { verbose :: !Bool,
+    -- For some reason, the Shake profile files are truncated and won't load
+    shakeProfiling :: !(Maybe FilePath),
```

Comment:
Odd. Any idea where they are truncated? Would be keen to debug this, but will wait til it lands.

Reply:
No idea. This might be a bug in ghcide